cdi/src/main/java/org/hibernate/validator/internal/cdi/ValidationExtension.java

```diff
@@ -277,6 +272,7 @@ else if ( bean.getTypes().contains( Validator.class ) || bean instanceof Validat
 				continue;
 			}
 
+			BeanDescriptor beanDescriptor = validator.getConstraintsForClass( method.getDeclaringClass() );
```

Comment:
I have created PR https://github.com/hibernate/hibernate-validator/pull/409 for that.